core/dbt/parser/unit_tests.py

```diff
@@ -147,23 +145,27 @@ def parse_unit_test_case(self, test_case: UnitTestDefinition):
                 "name": input_name,
                 "path": f"{input_name}.sql",
             }
+            resource_type = original_input_node.resource_type
 
-            if original_input_node.resource_type in (
+            if resource_type in (
                 NodeType.Model,
                 NodeType.Seed,
                 NodeType.Snapshot,
             ):
+
                 input_node = ModelNode(
                     **common_fields,
                     defer_relation=original_input_node.defer_relation,
                 )
-                if (
-                    original_input_node.resource_type == NodeType.Model
-                    and original_input_node.version
-                ):
+
+                if not resource_type == NodeType.Model:
+                    continue
```

Comment:
This is a little tricky to read given we're already in a `resource_type in (NodeType.Model,NodeType.Seed, NodeType.Snapshot)` block.

I'd suggest refactoring to be more straightforward to read, something like: 

```python
if resource_type == NodeType.model:
  input_node = ModelNode(
    **common_fields,
    defer_relation=original_input_node.defer_relation,
    latest_version=original_input_node.latest_version,
    version=original_input_node.version
  )
elif resource_type in (NodeType.Seed, NodeType.Snapshot):
  ...
elif resource_type == NodeType.Source:
  ... 
```

Reply:
done